pkg/policy/selectorcache.go

```diff
@@ -837,14 +845,19 @@ func (sc *SelectorCache) AddFQDNSelector(user CachedSelectionUser, fqdnSelec api
 		sc.selectors[key] = newFQDNSel
 	}
 
-	identitiesToRelease := newFQDNSel.transferIdentityReferencesToSelector(currentlyAllocatedIdentities)
-	ctx, cancel := context.WithTimeout(context.TODO(), option.Config.KVstoreConnectivityTimeout)
-	defer cancel()
-	sc.idAllocator.ReleaseCIDRIdentitiesByID(ctx, identitiesToRelease)
-
+	identitiesToRelease = newFQDNSel.transferIdentityReferencesToSelector(currentlyAllocatedIdentities)
 	newFQDNSel.updateSelections()
 
-	return newFQDNSel, newFQDNSel.addUser(user)
+	return newFQDNSel, newFQDNSel.addUser(user), identitiesToRelease
+}
+
+// AddFQDNSelector adds the given api.FQDNSelector in to the selector cache. If
+// an identical EndpointSelector has already been cached, the corresponding
+// CachedSelector is returned, otherwise one is created and added to the cache.
+func (sc *SelectorCache) AddFQDNSelector(user CachedSelectionUser, fqdnSelec api.FQDNSelector) (cachedSelector CachedSelector, added bool) {
```

Comment:
:shrug: :tomato: :tomato: 

For anything longer than a trivial function like this, I think we should *always* include the return values to improve readability. For shorter functions like this, I'm not fussed - both ways are valid Go syntax and Go isn't too complicated a language to learn, so most readers should understand either way. Strictly speaking there's no need to use named return values here, but since one of the values represents a boolean which could have different semantic meanings, the named return values act as documentation here. Arguably these return values are shadowed by the `:=` just below, which could just be `=`, but I don't know if there's any meaningful difference to iterating further on this.